activerecord/lib/active_record/migration/compatibility.rb

```diff
@@ -71,6 +71,19 @@ class << t
           end
         end
 
+        def create_join_table(table_1, table_2, column_options: {}, **options)
+          if block_given?
+            super(table_1, table_2, column_options: column_options.reverse_merge!(type: :integer), **options) do |t|
+              class << t
+                prepend TableDefinition
+              end
+              yield t
+            end
+          else
+            super(table_1, table_2, column_options: column_options.reverse_merge!(type: :integer), **options)
```

Comment:
We can simply call `super` if `column_options.reverse_merge!(type: :integer)` before if-else block.

Reply:
Done it. Thanks.